### ISSUE TITLE
Problem with dataset.num_classes

### ISSUE THREAD

**Issue by mesax1** [rusty1s/pytorch_geometric#1323]
## ❓ Questions & Help

Greetings,

I've recently started working with Pytorch Geometric,  and I'm trying to create and use my own dataset to do graph classification. I've followed different tutorials, such as https://towardsdatascience.com/hands-on-graph-neural-networks-with-pytorch-pytorch-geometric-359487e221a8 and the one in the official documentation https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html. 

I think everything works fine ( I suppose), except for the number of classes that are "assigned" to the dataset. My labels are 1 and 0, but I don't understand why num_classes is 24. 

The part of the code concerning the dataset creation is here: 
``
def to_pytorch_graph(self, z):
        dist_matrix = self.instance.dist_matrix
        edge_list = []
        edge_features = []
        node_features = [None] * len(self.customers)

        for customer in self.customers:
            node_features[customer.id] = [customer.demand]

        for vehicle in self.vehicles:
            curr_customer = -1
            prev_customer = -1

            for customer in vehicle.customers:
                curr_customer = customer.id
                
                if prev_customer != -1:
                    edge_list.append([prev_customer, curr_customer])
                    edge_features.append([dist_matrix[prev_customer][curr_customer]])

                prev_customer = curr_customer

        edge_index = torch.tensor(edge_list, dtype=torch.long).t().contiguous()
        x = torch.tensor(node_features, dtype=torch.float)
        edge_attr = torch.tensor(edge_features, dtype=torch.float)
        y = torch.tensor(z.reshape(1,-1), dtype=torch.float)
        #y = torch.tensor([z], dtype=torch.float)
        #y=z

        return Data(x=x, edge_index=edge_index, edge_attr=edge_attr, y=y)
for i in range(len(X)):
    X[i] = X[i].to_pytorch_graph(Z[i])

class MyOwnDataset(InMemoryDataset):
    def __init__(self, root, transform=None, pre_transform=None):
        super(MyOwnDataset, self).__init__(root, transform, pre_transform)
        self.data, self.slices = torch.load(self.processed_paths[0])

    @property
    def raw_file_names(self):
        return []

    @property
    def processed_file_names(self):
        return ['data.pt']

    def download(self):
        # Download to `self.raw_dir`.
        pass

    def process(self):
        
        data = X
        data, slices = self.collate(data)
        torch.save((data, slices), self.processed_paths[0])

dataset = MyOwnDataset(root='../',)

Could you please help me out? What am I doing wrong?

Thank you

**Comment by aishwaryyasarkar** [comment]
I have the same question. Did you find out the answer since I see that you closed this thread?

**Comment by rusty1s** [comment]
`num_classes` will try to infer the number of classes, but it might not yield the correct number in case you use some special encoding for labels. In general, it computes `num_classes` as follows:
1. In case `data.y` is one-dimensional, return `data.y.max() + 1`
2. In case `data.y` is multi-dimensional, return `data.size(-1)`.
